--- a/include/linux/mount.h
+++ b/include/linux/mount.h
@@ -69,6 +69,9 @@
 struct vfsmount {
 	struct dentry *mnt_root;	/* root of the mounted tree */
 	struct super_block *mnt_sb;	/* pointer to superblock */
+	int mnt_flags;
+	struct mnt_idmap *mnt_idmap;
+
 	ANDROID_KABI_RESERVE(1);
 	ANDROID_KABI_RESERVE(2);
 	ANDROID_KABI_RESERVE(3);
@@ -76,6 +79,12 @@
 	void *data;
 } __randomize_layout;
 
+static inline struct mnt_idmap *mnt_idmap(const struct vfsmount *mnt)
+{
+	/* Pairs with smp_store_release() in do_idmap_mount(). */
+	return smp_load_acquire(&mnt->mnt_idmap);
+}
+
 struct file; /* forward dec */
 struct path;
 
